Added Utility API functions

JABillard opened the pull request:
Added utility API functions and slight modifications to highlight_data.

JABillard committed:
Added various utility API functions and added an attribute to highlight_data which allows the user to use connection names as the column names for the given dataframe.

JABillard committed:
Slight edit to commenting

ttricco commented:
1. Convert all the id's to integers instead of strings.
2. get_function_resources(name, id) etc should be a single parameter, i.e., get_function_resources(function). If function is an integer, than treat it like an id, else if a string, treat like a function name.

ttricco commented:
Throw errors if inputs are invalid. For example, 

if isinstance(id, int):
 ...
elif isinstance(id, str):
 ...
else:
  raise ValueError

JABillard committed:
- Modified the storage of function ID's so that they are stored as integers rather than strings.

- Adjusted the "get_function_**aspect**s" functions so that they only use one parameter. Depending on if the value is an integer or string, determines if the program treats that value as a function ID (int) or name (str).

- Added various raise errors in the case that an invalid input is given.

- Adjusted the appearance of grey dashed lines when data is highlighted so that it doesn't cover highlighted lines. Instead, grey dashed lines will appear behind colored lines. This makes it easier for the user to see the highlighted lines, which is far mor important.

- Some adjustments to documentation.
